src/Microsoft.TestPlatform.CommunicationUtilities/LengthPrefixCommunicationChannel.cs

```diff
@@ -85,14 +85,10 @@ public Task NotifyDataAvailable()
             // Try read data even if no one is listening to the data stream. Some server
             // implementations (like Sockets) depend on the read operation to determine if a
             // connection is closed.
-            if (MessageReceived != null)
+            if (MessageReceived.WaitForSubscriber(1000, cancellationToken))
             {
                 var data = _reader.ReadString();
-                MessageReceived.SafeInvoke(this, new MessageReceivedEventArgs { Data = data }, "LengthPrefixCommunicationChannel: MessageReceived");
-            }
-            else
-            {
```

Comment:
This message is unnecessary, and way too much detail. with prestart it is normal that client is waiting for his turn and this would just add yet another message to the log that would always repeat.